### ISSUE TITLE
read_pickle support

### ISSUE THREAD

**Issue by weidinger-c** [rapidsai/cudf#16048]
cuDF currently has no IO support for pickle. I would need this function. 

`AttributeError: module 'cudf' has no attribute 'read_pickle'`

**Comment by weidinger-c** [comment]
Thanks, I know that there is a dedicated pickle moudle. 
I just that I wanted to compare my code without any code changes as I thought that cuDF has feature parity with pandas df.

**Comment by wence-** [comment]
> I just wanted to compare my code without any code changes as I thought that cuDF has feature parity with pandas df.

Mostly, but not completely. Other than the API compatibility is there some aspect of `pandas.read_pickle` that is not supported by plain `pickle.load`?

**Comment by weidinger-c** [comment]
> > I just wanted to compare my code without any code changes as I thought that cuDF has feature parity with pandas df.
> 
> Mostly, but not completely. Other than the API compatibility is there some aspect of `pandas.read_pickle` that is not supported by plain `pickle.load`?

No, at least nothing I am aware. As I said, I just wanted to try out and test my lib with cudf with the least possible effort to see if it brings some performance gains.
